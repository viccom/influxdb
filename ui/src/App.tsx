import React, {SFC, ReactChildren} from 'react'

import RightClickLayer from 'src/clockface/components/right_click_menu/RightClickLayer'
import Nav from 'src/pageLayout'
import Notifications from 'src/shared/components/notifications/Notifications'
import LegendPortal from 'src/shared/components/LegendPortal'

interface Props {
  children: ReactChildren
}

const App: SFC<Props> = ({children}) => (
  <div className="chronograf-root">
    <Notifications />
    <RightClickLayer />
    <Nav />
    <LegendPortal />
    {children}
  </div>
)

export default App
