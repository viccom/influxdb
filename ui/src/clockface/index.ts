// Import Components
import AutoInput from './components/auto_input/AutoInput'
import Button from './components/Button'
import ConfirmationButton from './components/confirmation_button/ConfirmationButton'
import Dropdown, {DropdownMode} from './components/dropdowns/Dropdown'
import MultiSelectDropdown from './components/dropdowns/MultiSelectDropdown'
import Form from './components/form_layout/Form'
import Input, {InputType, AutoComplete} from './components/inputs/Input'
import MultipleInput, {
  MultiInputType,
} from './components/inputs/multipleInput/MultipleInput'
import OverlayTechnology from './components/overlays/OverlayTechnology'
import OverlayContainer from './components/overlays/OverlayContainer'
import OverlayHeading from './components/overlays/OverlayHeading'
import OverlayBody from './components/overlays/OverlayBody'
import OverlayFooter from './components/overlays/OverlayFooter'
import Panel from './components/panel/Panel'
import Radio from './components/radio_buttons/RadioButtons'
import SlideToggle from './components/slide_toggle/SlideToggle'
import WizardFullScreen from './components/wizard/WizardFullScreen'
import WizardOverlay from './components/wizard/WizardOverlay'
import WizardProgressHeader from './components/wizard/WizardProgressHeader'
import ProgressBar from './components/wizard/ProgressBar'
import ComponentSpacer from './components/component_spacer/ComponentSpacer'
import EmptyState from './components/empty_state/EmptyState'
import Spinner from './components/spinners/Spinner'
import SparkleSpinner from './components/spinners/SparkleSpinner'
import IndexList from './components/index_views/IndexList'
import Context from './components/context_menu/Context'
import FormElement from 'src/clockface/components/form_layout/FormElement'
import DraggableResizer from 'src/clockface/components/draggable_resizer/DraggableResizer'
import Label, {LabelType} from 'src/clockface/components/label/Label'
import LabelSelector from 'src/clockface/components/label/LabelSelector'
import GridSizer from 'src/clockface/components/grid_sizer/GridSizer'
import Select from 'src/clockface/components/Select'
import Grid from 'src/clockface/components/grid_layout/Grid'

// Import Types
import {
  ComponentColor,
  ComponentSize,
  ComponentStatus,
  DropdownMenuColors,
  DropdownChild,
  ButtonShape,
  Greys,
  IconFont,
  Columns,
  Alignment,
  Stack,
  ButtonType,
  Sort,
} from './types'

// Fire de lazer
export {
  Alignment,
  AutoComplete,
  AutoInput,
  Button,
  ButtonType,
  ButtonShape,
  Columns,
  ComponentColor,
  ComponentSize,
  ComponentSpacer,
  ComponentStatus,
  ConfirmationButton,
  Context,
  DraggableResizer,
  Dropdown,
  DropdownChild,
  DropdownMenuColors,
  DropdownMode,
  EmptyState,
  Form,
  FormElement,
  Greys,
  Grid,
  GridSizer,
  IconFont,
  IndexList,
  Input,
  InputType,
  Label,
  LabelSelector,
  LabelType,
  MultiSelectDropdown,
  MultiInputType,
  MultipleInput,
  OverlayBody,
  OverlayContainer,
  OverlayFooter,
  OverlayHeading,
  OverlayTechnology,
  Panel,
  ProgressBar,
  Radio,
  Select,
  SlideToggle,
  SparkleSpinner,
  Sort,
  Spinner,
  Stack,
  WizardFullScreen,
  WizardProgressHeader,
  WizardOverlay,
}
