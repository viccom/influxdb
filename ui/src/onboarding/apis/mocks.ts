import {
  getTelegrafConfigsResponse,
  authResponse,
  createTelegrafConfigResponse,
  setSetupParamsResponse,
} from 'mocks/dummyData'

const telegrafsGet = jest.fn(() => Promise.resolve(getTelegrafConfigsResponse))
const telegrafsPost = jest.fn(() =>
  Promise.resolve(createTelegrafConfigResponse)
)
const telegrafsTelegrafIDPut = jest.fn(() =>
  Promise.resolve(createTelegrafConfigResponse)
)
const authorizationsGet = jest.fn(() => Promise.resolve(authResponse))
const setupPost = jest.fn(() => Promise.resolve(setSetupParamsResponse))
const setupGet = jest.fn(() => Promise.resolve({data: {allowed: true}}))

export const telegrafsAPI = {
  telegrafsGet,
  telegrafsPost,
  telegrafsTelegrafIDPut,
}

export const setupAPI = {
  setupPost,
  setupGet,
}

export const authorizationsAPI = {
  authorizationsGet,
}
