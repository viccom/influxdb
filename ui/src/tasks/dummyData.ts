import {Task} from 'src/api'

export const dummyTasks: Task[] = [
  {
    id: '02ef9deff2141000',
    orgID: '02ee9e2a29d73000',
    name: 'pasdlak',
    status: Task.StatusEnum.Active,
    owner: null,
    flux:
      'option task = {\n  name: "pasdlak",\n  cron: "2 0 * * *"\n}\nfrom(bucket: "inbucket") \n|> range(start: -1h)',
    cron: '2 0 * * *',
  },
  {
    id: '02f12c50dba72000',
    orgID: '02ee9e2a29d73000',
    name: 'somename',
    status: Task.StatusEnum.Active,
    owner: null,
    flux:
      'option task = {\n  name: "somename",\n  every: 1m,\n}\nfrom(bucket: "inbucket") \n|> range(start: -task.every)',
    every: '1m0s',
  },
]
