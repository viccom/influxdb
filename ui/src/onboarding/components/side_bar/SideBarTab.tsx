// Libraries
import React, {Component} from 'react'
import classnames from 'classnames'

// Types
import {IconFont} from 'src/clockface'
import {SideBarTabStatus as TabStatus} from 'src/onboarding/components/side_bar/SideBar'

interface Props {
  label: string
  id: string
  active: boolean
  status: TabStatus
  onClick: (tabID: string) => void
}

class SideBarTab extends Component<Props> {
  public render() {
    const {label} = this.props

    return (
      <div className={this.className} onClick={this.handleClick}>
        {this.icon}
        {label}
      </div>
    )
  }

  private handleClick = (): void => {
    const {id, onClick} = this.props

    onClick(id)
  }

  private get className(): string {
    const {status, active} = this.props

    return classnames('side-bar--tab', {
      active,
      'side-bar--tab__success': status === TabStatus.Success,
      'side-bar--tab__error': status === TabStatus.Error,
      'side-bar--tab__pending': status === TabStatus.Pending,
    })
  }

  private get icon(): JSX.Element {
    const {status} = this.props
    let icon

    switch (status) {
      case TabStatus.Pending:
      case TabStatus.Success:
        icon = `side-bar--icon icon ${IconFont.Checkmark}`
        break
      case TabStatus.Error:
        icon = `side-bar--icon icon ${IconFont.Remove}`
        break
      case TabStatus.Default:
        icon = `side-bar--icon icon ${IconFont.CircleThick}`
        break
      default:
        icon = `side-bar--icon`
    }

    return <span className={icon} />
  }
}

export default SideBarTab
