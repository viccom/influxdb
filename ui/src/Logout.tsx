// Libraries
import {PureComponent} from 'react'
import {withRouter, WithRouterProps} from 'react-router'

// APIs
import {logout} from 'src/me/apis'

// Components
import {ErrorHandling} from 'src/shared/decorators/errors'

type Props = WithRouterProps

@ErrorHandling
export class Logout extends PureComponent<Props> {
  public componentDidMount() {
    this.handleSignOut()
  }

  public render() {
    return null
  }

  private handleSignOut = async () => {
    await logout()
    this.props.router.push('/')
  }
}

export default withRouter<Props>(Logout)
