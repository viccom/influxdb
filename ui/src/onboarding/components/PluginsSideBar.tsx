import React, {Component} from 'react'

// Components
import SideBar from 'src/onboarding/components/side_bar/SideBar'
import {SideBarTabStatus as TabStatus} from 'src/onboarding/components/side_bar/SideBar'
import {ComponentColor} from 'src/clockface'

// Utils
import {downloadTextFile} from 'src/shared/utils/download'

// Constants
import {getTelegrafConfigFailed} from 'src/shared/copy/v2/notifications'

// APIs
import {getTelegrafConfigTOML} from 'src/onboarding/apis'

// Types
import {IconFont} from 'src/clockface'
import {TelegrafPlugin, ConfigurationState} from 'src/types/v2/dataLoaders'
import {NotificationAction} from 'src/types'

interface Props {
  title: string
  telegrafConfigID: string
  visible: boolean
  telegrafPlugins: TelegrafPlugin[]
  notify: NotificationAction
  onTabClick: (tabID: string) => void
  currentStepIndex: number
  onNewSourceClick: () => void
}

const configStateToTabStatus = (cs: ConfigurationState): TabStatus => {
  switch (cs) {
    case ConfigurationState.Unconfigured:
      return TabStatus.Default
    case ConfigurationState.InvalidConfiguration:
      return TabStatus.Error
    case ConfigurationState.Configured:
      return TabStatus.Success
  }
}

class PluginsSideBar extends Component<Props> {
  public render() {
    const {title, visible} = this.props
    return (
      <SideBar title={title} visible={visible}>
        {this.content}
      </SideBar>
    )
  }
  private get content(): JSX.Element[] {
    const {currentStepIndex} = this.props
    if (currentStepIndex !== 2) {
      return [...this.tabs, ...this.buttons]
    }
    return this.tabs
  }

  private get tabs(): JSX.Element[] {
    const {telegrafPlugins, onTabClick} = this.props
    return telegrafPlugins.map(t => (
      <SideBar.Tab
        label={t.name}
        key={t.name}
        id={t.name}
        active={t.active}
        status={configStateToTabStatus(t.configured)}
        onClick={onTabClick}
      />
    ))
  }

  private get downloadButton(): JSX.Element {
    return (
      <SideBar.Button
        key="Download Config File"
        text="Download Config File"
        titleText="Download Config File"
        data-test="download"
        color={ComponentColor.Secondary}
        icon={IconFont.Download}
        onClick={this.handleDownload}
      />
    )
  }

  private get addSourceButton(): JSX.Element {
    const {onNewSourceClick} = this.props

    return (
      <SideBar.Button
        text="Add New Source"
        key="Add New Source"
        titleText="Add New Source"
        data-test="new"
        color={ComponentColor.Default}
        icon={IconFont.Plus}
        onClick={onNewSourceClick}
      />
    )
  }

  private get buttons(): JSX.Element[] {
    const {telegrafConfigID} = this.props

    if (telegrafConfigID) {
      return [this.downloadButton, this.addSourceButton]
    }
    return [this.addSourceButton]
  }

  private handleDownload = async () => {
    const {notify, telegrafConfigID} = this.props
    try {
      const config = await getTelegrafConfigTOML(telegrafConfigID)
      downloadTextFile(config, 'config.toml')
    } catch (error) {
      notify(getTelegrafConfigFailed())
    }
  }
}

export default PluginsSideBar
