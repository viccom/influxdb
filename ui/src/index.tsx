import 'babel-polyfill'

import React, {PureComponent} from 'react'
import {render} from 'react-dom'
import {Provider} from 'react-redux'
import {Router, Route, useRouterHistory, IndexRoute} from 'react-router'
import {createHistory, History} from 'history'

import configureStore from 'src/store/configureStore'
import {loadLocalStorage} from 'src/localStorage'

import {getRootNode} from 'src/utils/nodes'
import {getBasepath} from 'src/utils/basepath'

// Components
import App from 'src/App'
import GetSources from 'src/shared/containers/GetSources'
import SetActiveSource from 'src/shared/containers/SetActiveSource'
import GetOrganizations from 'src/shared/containers/GetOrganizations'
import Setup from 'src/Setup'
import Signin from 'src/Signin'
import SigninPage from 'src/onboarding/containers/SigninPage'
import Logout from 'src/Logout'
import TaskPage from 'src/tasks/containers/TaskPage'
import TasksPage from 'src/tasks/containers/TasksPage'
import OrganizationsIndex from 'src/organizations/containers/OrganizationsIndex'
import OrganizationView from 'src/organizations/containers/OrganizationView'
import TaskEditPage from 'src/tasks/containers/TaskEditPage'
import DashboardPage from 'src/dashboards/components/DashboardPage'
import DashboardsIndex from 'src/dashboards/components/dashboard_index/DashboardsIndex'
import DataExplorerPage from 'src/dataExplorer/components/DataExplorerPage'
import {MePage, Account} from 'src/me'
import NotFound from 'src/shared/components/NotFound'
import GetLinks from 'src/shared/containers/GetLinks'
import GetMe from 'src/shared/containers/GetMe'
import SourcesPage from 'src/sources/components/SourcesPage'
import ConfigurationPage from 'src/configuration/components/ConfigurationPage'

import OnboardingWizardPage from 'src/onboarding/containers/OnboardingWizardPage'

// Actions
import {disablePresentationMode} from 'src/shared/actions/app'

// Styles
import 'src/style/chronograf.scss'

const rootNode = getRootNode()
const basepath = getBasepath()

declare global {
  interface Window {
    basepath: string
  }
}

// Older method used for pre-IE 11 compatibility
window.basepath = basepath

const history: History = useRouterHistory(createHistory)({
  basename: basepath, // this is written in when available by the URL prefixer middleware
})

const store = configureStore(loadLocalStorage(), history)
const {dispatch} = store

history.listen(() => {
  dispatch(disablePresentationMode())
})

window.addEventListener('keyup', event => {
  const escapeKeyCode = 27
  // fallback for browsers that don't support event.key
  if (event.key === 'Escape' || event.keyCode === escapeKeyCode) {
    dispatch(disablePresentationMode())
  }
})

class Root extends PureComponent {
  public render() {
    return (
      <Provider store={store}>
        <Router history={history}>
          <Route component={GetLinks}>
            <Route component={Setup}>
              <Route path="/onboarding">
                <Route path=":stepID" component={OnboardingWizardPage} />
                <Route
                  path=":stepID/:substepID"
                  component={OnboardingWizardPage}
                />
              </Route>
              <Route path="/signin" component={SigninPage} />
              <Route path="/logout" component={Logout} />
              <Route path="/">
                <Route component={Signin}>
                  <Route component={GetMe}>
                    <Route component={GetOrganizations}>
                      <Route component={App}>
                        <Route component={GetSources}>
                          <Route component={SetActiveSource}>
                            <IndexRoute component={DashboardsIndex} />
                            <Route path="organizations">
                              <IndexRoute component={OrganizationsIndex} />
                              <Route
                                path=":orgID/:tab"
                                component={OrganizationView}
                              />
                            </Route>
                            <Route path="tasks">
                              <IndexRoute component={TasksPage} />
                              <Route path="new" component={TaskPage} />
                              <Route path=":id" component={TaskEditPage} />
                            </Route>
                            <Route
                              path="data-explorer"
                              component={DataExplorerPage}
                            />
                            <Route path="dashboards">
                              <IndexRoute component={DashboardsIndex} />
                              <Route
                                path=":dashboardID"
                                component={DashboardPage}
                              />
                            </Route>
                            <Route path="me" component={MePage} />
                            <Route path="account/:tab" component={Account} />
                            <Route path="sources" component={SourcesPage} />
                            <Route
                              path="configuration/:tab"
                              component={ConfigurationPage}
                            />
                          </Route>
                        </Route>
                      </Route>
                    </Route>
                  </Route>
                </Route>
              </Route>
            </Route>
          </Route>
          <Route path="*" component={NotFound} />
        </Router>
      </Provider>
    )
  }
}

if (rootNode) {
  render(<Root />, rootNode)
}
