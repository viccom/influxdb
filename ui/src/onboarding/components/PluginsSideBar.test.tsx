// Libraries
import React from 'react'
import {shallow} from 'enzyme'

// Components
import PluginsSideBar from 'src/onboarding/components/PluginsSideBar'
import {cpuTelegrafPlugin, diskTelegrafPlugin} from 'mocks/dummyData'
import {Button} from 'src/clockface'
import SideBarTab from 'src/onboarding/components/side_bar/SideBarTab'

const onClick = jest.fn(() => {})

const setup = (override = {}) => {
  const props = {
    title: 'title',
    visible: true,
    telegrafPlugins: [],
    onTabClick: onClick,
    telegrafConfigID: '',
    notify: jest.fn(),
    currentStepIndex: 0,
    onNewSourceClick: jest.fn(),
    ...override,
  }

  const wrapper = shallow(<PluginsSideBar {...props} />)

  return {wrapper}
}

describe('PluginsSideBar', () => {
  describe('rendering', () => {
    it('renders! wee!', () => {
      const {wrapper} = setup({
        telegrafPlugins: [cpuTelegrafPlugin, diskTelegrafPlugin],
      })

      expect(wrapper.exists()).toBe(true)
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('if on selection step', () => {
    it('renders the tabs but no buttons', () => {
      const {wrapper} = setup({
        currentStepIndex: 2,
        telegrafPlugins: [cpuTelegrafPlugin, diskTelegrafPlugin],
      })

      const buttons = wrapper.find(Button)
      const tabs = wrapper.find(SideBarTab)

      expect(buttons.exists()).toBe(false)
      expect(tabs.exists()).toBe(true)
    })
  })

  describe('if there is not a telegraf config id', () => {
    it('renders the streaming sources button but not download config button', () => {
      const {wrapper} = setup({currentStepIndex: 3})

      const downloadButton = wrapper.find('[data-test="download"]')
      const addNewSourceButton = wrapper.find('[data-test="new"]')

      expect(downloadButton.exists()).toBe(false)
      expect(addNewSourceButton.exists()).toBe(true)
    })
  })

  describe('if there is a telegraf config id', () => {
    it('renders the streaming sources button and the download config button', () => {
      const {wrapper} = setup({
        currentStepIndex: 3,
        telegrafConfigID: 'ajefoiaijesfk',
      })

      const downloadButton = wrapper.find('[data-test="download"]')
      const addNewSourceButton = wrapper.find('[data-test="new"]')

      expect(downloadButton.exists()).toBe(true)
      expect(addNewSourceButton.exists()).toBe(true)
    })
  })
})
