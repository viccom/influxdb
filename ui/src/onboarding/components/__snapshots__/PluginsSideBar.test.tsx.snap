// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PluginsSideBar rendering renders! wee! 1`] = `
<SideBar
  title="title"
  visible={true}
>
  <SideBarTab
    active={false}
    id="cpu"
    key="cpu"
    label="cpu"
    onClick={[MockFunction]}
    status="success"
  />
  <SideBarTab
    active={false}
    id="disk"
    key="disk"
    label="disk"
    onClick={[MockFunction]}
    status="success"
  />
  <SideBarButton
    color="default"
    data-test="new"
    icon="plus"
    key="Add New Source"
    onClick={[MockFunction]}
    text="Add New Source"
    titleText="Add New Source"
  />
</SideBar>
`;
